src/components/subscribe/subscribe.stories.mdx

```diff
@@ -224,14 +242,21 @@ The Subscribe component `.js-subscribe` root element.
 
 ### Return value
 
-An object with a `destroy` function that removes all event listeners added by this component.
+An object with two functions to programmatically control the Subscribe component's
+state:
 
 ```js
 {
+  reinit: () => void
```

Comment:
This might just be me, but I find the concept of `reinit` really confusing.

I think in the past I've seen patterns like this...

```js
// Instantiate the component
const subscribe = SubscribeComponent(element);
// Destroy later
subscribe.destroy();
// Re-initialize
subscribe.init();
```

Sometimes they require an _initial_ `init` call, or sometimes they make the `init` call return the main object so you can use a shorthand like this:

```js
const subscribe = SubscribeComponent(element).init();
subscribe.destroy();
```

I believe the benefit of that approach is that you can optionally set up the thing in one step and then get it kicked off in another. But I couldn't say for sure.

The main reason I find it more intuitive is that `init` and `destroy` basically mean `on` and `off`. Wanna activate the component? That's `init`. Wanna deactivate it? That's `destroy`.

I don't know where `reinit` fits in there. It's a weird term, and there's no `init` method for me to compare it to.

I dunno. Maybe I've just got 4pm brain or something.

Reply:
Yeah, I actually had `init`/`destroy` but then I thought it might be odd that it initializes automatically when you run:
```js
const subscribe = initSubscribe(element);
```
So I thought, since it's already initialized, your options from there are to destroy it or reinitialize it. But, I'm fine switching it back from `reinit` to `init` ([`reinit` calls `init` under the hood anyways](https://github.com/cloudfour/cloudfour.com-patterns/pull/1882/files#diff-2cf35c939f52cd7cb90f679aef3a3bdb9d78236a9588414e3b49f41a006f218dR144)). 👍🏽 

Are you also suggesting the following change from:
```js
export const initSubscribe = (containerEl: HTMLElement) => { ... }
```
to
```js
export const SubscribeComponent = (containerEl: HTMLElement) => { ... }
```

Reply:
Additional info...

[Here is where `init()` is automatically called](https://github.com/cloudfour/cloudfour.com-patterns/blob/c1df9aff13616bfc0720ff01829a100645085b4e/src/components/subscribe/subscribe.ts#L101) when you call `initSubscribe`.

And regarding only exporting the single `init*()` function, I was following [existing patterns](https://github.com/cloudfour/cloudfour.com-patterns/blob/c1df9aff13616bfc0720ff01829a100645085b4e/src/components/sky-nav/sky-nav.ts#L14) already set up in the repo. 

I don't mind switching to this pattern:
```js
// Instantiate the component
const subscribe = SubscribeComponent(element);
// Destroy later
subscribe.destroy();
// Re-initialize
subscribe.init();
```
But we should probably do a sweep of all JS components so that there is consistency across all of their APIs. Is it safe to assume `SubscribeComponent()` does not run `init()`?

Reply:
Ultimately the most important thing is that you and your @cloudfour/dev collaborators are all on board. I just wanted to poke at this because sometimes in a team it can be tough for others to know what is or isn't up for discussion if there isn't someone swooping and pooping on certain details from time to time. 😅

Reply:
I agree that the implicit initialization being a separate thing than `reinit` might be a little confusing.

Can we call the first function `createSubscribe`?

```
// Instantiate the component
const subscribe = createSubscribe(element);
subscribe.init();

// Destroy later
subscribe.destroy();
// Re-initialize
subscribe.init();
```

Reply:
One more thought...

> But we should probably do a sweep of all JS components so that there is consistency across all of their APIs.

In our client work, we often have to think about the best structure and consistency for the customer. Often that involves accommodating some work they've already done to that point. In this case, _we're_ the client. Any conventions we're following are our own, and we should feel empowered to change them, or to break consistency for one component and make an issue to refactor the others later, or whatever makes sense for our values and workflow.

I have a feeling that in this case, @calebeby's suggestion makes a lot of sense. I'm not suggesting that we need to balloon this task into a full-on refactor, just making sure folks remember the power we have in this particular space.

Reply:
I think using the existing API pattern in other components, if we wanted to be able destroy them and initialize them again would be like this:

```js
let something = initializeWhatever(element);
// ...later
something.destroy();
// ...later re-initialize
something = initializeWhatever(element);
```

But, I don't think for other components there is really the use-case for wanting to destroy and reinitialize them multiple times.

For the subscribe component, uninitializing and reinitializing is more of a common workflow. I would be fine with either way, calling the top-level method multiple times to reinitialize, or calling a returned `init` method each time we want to initialize.

Because the usage pattern of this component is pretty different from other components, I don't think there is a strong case that the API needs to be exactly consistent. I don't think seeing the `init` function on just this one component would be too out of place.

Reply:
> But we should probably do a sweep of all JS components so that there is consistency across all of their APIs.

> I'm not suggesting that we need to balloon this task into a full-on refactor, just making sure folks remember the power we have in this particular space.

If you'll pardon me jumping in, I think this is a point where I have some useful experience working on product teams for longer-term products.

When a team discovers a better way to do things they've been doing, they need to avoid falling into either of these traps:

- Sunk-cost fallacy: We can't change this now, we've already done it this way in too many places.
- Burn it all down: If we change it here, we have to change it everywhere.

No one likes it when we change a standard or best practice and then the code base has a split of files using two techniques. That said, I've participated in several large-scale refactoring efforts of long-term projects. In these teams, there simply isn't a feasable way to refactor all existing code at once, because the effort required would block all ability to do any other work. But you can't be unwilling to change best practices, either. In those teams, it's really common to communicate changes out to the team, and make it clear "this is how we do this moving forward, and if you touch old code, you should take the time to update it."

Ironically, because this codebase is our baby, it muddies the waters. When it's a client project, we usually have a firm deadline or budget driving us, so it's clear when a refactoring effort would blow our scope. But since this is our project on our deadline, it's harder to make a decision.

Thankfully, in this case, I would argue we _do_ have a deadline. We have a mandate from Megan to burn towards releasing the new site as soon as possible. For me, that says clearly that we should not attempt to go back and refactor old code, but we should note and share the new standard with the team, apply that standard to all PRs moving forward (including this one), and file tickets (which can be prioritized after launch) for refactoring certain pain points that we may already be aware of.

Apologies if everyone was already on the same page, but this was buzzing around my head last night and I wanted to share it with the team.

Reply:
> Can we call the first function `createSubscribe`?

This works for me!

> Because the usage pattern of this component is pretty different from other components, I don't think there is a strong case that the API needs to be exactly consistent. I don't think seeing the init function on just this one component would be too out of place.

Yep, I agree. This is how I started feeling about this overnight as well. 

I'm going to:
- rename `initSubscribe` to `createSubscribe`
- remove the implicit initilization
- change `reinit` to `init`